src/pudl/output/pudltabl.py

```diff
@@ -129,6 +131,39 @@ def __init__(
         # Used to persist the output tables. Returns None if they don't exist.
         self._dfs = defaultdict(lambda: None)
 
+        self._register_output_methods()
+
+    def _register_output_methods(self):
+        """Load output assets and register a class method for retrieving each one."""
+        output_assets = (
+            *load_assets_from_modules([pudl.output.denorm_ferc1]),
+            *load_assets_from_modules([pudl.output.denorm_eia]),
+            *load_assets_from_modules([pudl.transform.ferc714]),
+            *load_assets_from_modules([pudl.transform.eia861]),
+        )
+
+        package = Package.from_resource_ids()
+        for key in pudl.helpers.get_asset_keys(output_assets):
+            asset_name = key.to_python_identifier()
+
+            # Create method called asset_name that will read the asset from DB
+            self.__dict__[asset_name] = partial(
+                self._get_table_from_db,
+                table_name=asset_name,
+                resource=package.get_resource(asset_name),
+            )
```

Comment:
Instead of creating the entire package, you could also call `Resource.from_id(asset_name)`.

Reply:
What do you think is best @zaneselvans? I feel like there are other places in the code base where we create a Package with all of the default resources to get one resource object.

Reply:
I went ahead and changed it to `Resource.from_id`, seems better

Reply:
I always default to creating a Package because Pydantic is very fast (compared to everything else we do) and there are some kinds of metadata that only get created when the whole Package is instantiated -- in particular the foreign key relationships are only built by the Package and not by the Resource, and I think I ran into problems attempting use a Resource that wasn't part of a Package to enforce the schemas in the FERC 1 tables.

Reply:
The `Package` could be created upon instantiation of the `PudlTabl` object rather than inside the function, and then you could just use `self._package.get_resource()` rather than creating a new `Package` from scratch.